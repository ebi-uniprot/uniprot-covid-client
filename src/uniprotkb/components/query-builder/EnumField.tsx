import React, { Fragment } from 'react';
import { SearchTermType } from '../../types/searchTypes';

type EnumFieldProps = {
  field: SearchTermType;
  handleChange: (value: string) => void;
  value: string | undefined;
};

const EnumField: React.FC<EnumFieldProps> = ({
  field,
  handleChange,
  value,
}) => (
  <Fragment>
    <label htmlFor={`select_${field.term}`}>
      {field.label}
      <select
        onChange={e => handleChange(e.target.value)}
        id={`select_${field.term}`}
        data-testid="enum-field-select"
        value={value}
      >
        {field.values &&
          field.values.map(item => (
            <option value={item.value} key={`select_${item.value}`}>
              {item.name}
            </option>
          ))}
      </select>
    </label>
  </Fragment>
);

export default EnumField;
