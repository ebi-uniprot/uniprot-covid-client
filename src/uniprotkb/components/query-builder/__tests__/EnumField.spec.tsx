import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import EnumField from '../EnumField';

const props = {
  field: {
    label: 'Protein Existence [PE]',
    itemType: 'single',
    term: 'existence',
    dataType: 'enum',
    values: [
      {
        name: 'Evidence at protein level',
        value: '1',
      },
      {
        name: 'Evidence at transcript level',
        value: '2',
      },
    ],
    description: 'Search by protein existence',
    example: '1',
  },
  handleChange: jest.fn(),
  queryInput: { stringValue: '1' },
};

describe('Enum field', () => {
  test('should render an enum field', () => {
    const { asFragment, getByTestId } = render(<EnumField {...props} />);
    const select = getByTestId('enum-field-select');
    expect(select.value).toBe('1');
    fireEvent.change(select, { target: { value: '2' } });
    expect(select.value).toBe('2');
    expect(props.handleChange).toBeCalled();
    expect(asFragment()).toMatchSnapshot();
  });
});
