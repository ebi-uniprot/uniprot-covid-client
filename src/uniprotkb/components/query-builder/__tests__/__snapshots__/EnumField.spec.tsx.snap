// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Enum field should render an enum field 1`] = `
<DocumentFragment>
  <label
    for="select_existence"
  >
    Protein Existence [PE]
    <select
      data-testid="enum-field-select"
      id="select_existence"
    >
      <option
        value="1"
      >
        Evidence at protein level
      </option>
      <option
        value="2"
      >
        Evidence at transcript level
      </option>
    </select>
  </label>
</DocumentFragment>
`;
